src/RouteAggregator.js

```diff
@@ -0,0 +1,132 @@
+import invariant from 'invariant';
+import Relay from 'react-relay';
+
+import getParamsForRoute from './getParamsForRoute';
+
+export default class RouteAggregator {
+  constructor() {
+    this.uniqueQueryNames = new WeakMap();
```

Comment:
Only with `babel-runtime`. It's fantastic for libraries because it lets you polyfill things that should be treated as globals without polluting the global namespace.